src/main/java/uk/co/onsdigital/discovery/metadata/api/model/DimensionOption.java

```diff
@@ -22,9 +30,22 @@ public String getName() {
         return name;
     }
 
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    public Set<DimensionOption> getOptions() {
+        return options;
+    }
+
+    public void addOption(DimensionOption option) {
+        this.options.add(option);
+    }
+
     @Override
     public int compareTo(DimensionOption that) {
-        return Objects.compare(this.name, that.name, String.CASE_INSENSITIVE_ORDER);
+        return ComparisonChain.start()
+                .compare(this.id, that.id, String.CASE_INSENSITIVE_ORDER)
+                .compare(this.name, that.name, String.CASE_INSENSITIVE_ORDER)
+                .compare(this.options, that.options, Ordering.natural().lexicographical())
+                .result();
```

Comment:
It is for the ordering in the API, this is actually just this complicated so that the unit tests are short and readable. I could extract that into a separate comparator just in the tests I suppose.